### PR TITLE
feat: add C implementation for `stats/base/dists/uniform/mean`

### LINKED ISSUE #3830: [RFC]: Add C implementation for `@stdlib/stats/base/dists/uniform/mean`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/uniform/mean`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/uniform/mean) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/uniform/mean" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/mean/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/uniform/mean/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by 0PrashantYadav0**
Resolves #3830 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/uniform/mean along with relevant tests, benchmarks and examples

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3830

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 1]
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
...
+    double a;
+    double b;
+    double y;
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        a = (double)( rand() % 10 );
+        b = a + (double)( rand() % 10 + 1 ); // ensure b > a
+        y = stdlib_base_dists_uniform_mean( a, b );
+        printf( "a: %lf, b: %lf, Mean(X;a,b): %lf\n", a, b, y );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
        printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/benchmark.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2018 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2018 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2018 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
Copyright (c) 2018 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 4]
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/uniform/mean.h"
+#include <stdlib.h>
+#include <stdio.h>
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md:202]: You mind adding

```
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}
```

here and using it to draw the random numbers here, please? Applies to the example and benchmark C files, too.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for the PR!

Left a bunch of comments we should address.

I believe the slash commands didn't work for you because any workflows for new contributors have to initially be approved. I will look into us making this clearer...

**Comment by 0PrashantYadav0** [comment]
@Planeshifter Make all the changes. Please review all the changes.

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/src/main.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+*/
+
+#include "stdlib/stats/base/dists/uniform/mean.h"
+#include "stdlib/math/base/assert/is_nan.h"
+
+/**
+* Returns the mean of a uniform distribution.
+*
+* @param a    lower bound
+* @param b    upper bound
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/src/main.c]:
```suggestion
* @param a    minimum support
* @param b    maximum support
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/include/stdlib/stats/base/dists/uniform/mean.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Returns the mean of a uniform distribution with lower bound `a` and upper bound `b`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/include/stdlib/stats/base/dists/uniform/mean.h]:
```suggestion
* Returns the mean of a uniform distribution with minimum support `a` and maximum support `b`.
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/c/benchmark.c [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+*
+* @return elapsed time in seconds
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double a[ 100 ];
+	double b[ 100 ];
+	double y;
+	double t;
+	int i;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/benchmark/c/benchmark.c]:
```suggestion
	double a[ 100 ];
	double b[ 100 ];
	double elapsed;
	double y;
	double t;
	int i;
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 8]
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
...
+### Usage
+
+```c
+#include "stdlib/stats/base/dists/uniform/mean.h"
+```
+
+#### stdlib_base_dists_uniform_mean( a, b )
+
+Returns the [mean][mean] of a [uniform][uniform-distribution] distribution with minimum value \( a \) and maximum value \( b \).
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
Returns the [expected value][expected-value] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 9]
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
...
+
+```c
+double out = stdlib_base_dists_uniform_mean( 0.0, 1.0 );
+// returns 0.5
+```
+
+The function accepts the following arguments:
+
+-   **a**: `[in] double` minimum value.
+-   **b**: `[in] double` maximum value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
-   **a**: `[in] double` minimum support.
-   **b**: `[in] double` maximum support.
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 10]
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
...
+    double y;
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        a = (double)( rand() % 10 );
+        b = a + (double)( rand() % 10 + 1 ); // ensure b > a
+        y = stdlib_base_dists_uniform_mean( a, b );
+        printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
+    }
+}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double a;
    double b;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        a = random_uniform( 0.0, 10.0 );
        b = random_uniform( a, 20.0 ); // ensure b > a
        y = stdlib_base_dists_uniform_mean( a, b );
        printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
    }
}
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/examples/c/example.c [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+	double a;
+	double b;
+	double y;
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		a = random_uniform( 0.0, 10.0 );
+		b = random_uniform( a, 20.0 );
+		y = stdlib_base_dists_uniform_mean( a, b );
+		printf( "a: %lf , b: %lf , Mean: %lf\n", a, b, y );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/examples/c/example.c]:
```suggestion
		printf( "a: %lf , b: %lf , E(X;a,b): %lf\n", a, b, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/lib/native.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+
+// MAIN //
+
+/**
+* Returns the mean of a uniform distribution.
+*
+* @private
+* @param {number} a - lower bound
+* @param {number} b - upper bound
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/lib/native.js]:
```suggestion
* @param {number} a - minimum support
* @param {number} b - maximum support
```

**Review by Planeshifter — COMMENTED** [review]
Thanks for the PR!

Made a bunch of fixes, but should be ready for landing soon.

File: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md [hunk 13]
```diff
@@ -155,6 +250,8 @@ for ( i = 0; i < 10; i++ ) {
 
 [uniform-distribution]: https://en.wikipedia.org/wiki/Uniform_distribution_%28continuous%29
 
+[mean]: https://en.wikipedia.org/wiki/Mean
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md]:
```suggestion
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**